packages/cli/src/commands/hydrogen/build.ts

```diff
@@ -78,11 +51,13 @@ export default class Build extends Command {
       directory,
     };
 
-    if (await hasViteConfig(directory ?? process.cwd())) {
-      const {runViteBuild} = await import('./build-vite.js');
-      await runViteBuild(buildParams);
-    } else {
+    if (await hasViteConfig(directory)) {
       await runBuild(buildParams);
+    } else {
+      const {runClassicCompilerBuild} = await import(
+        '../../lib/classic-compiler/build.js'
+      );
```

Comment:
Is the reason to only lazy load the classic compiler because we'll eventually deprecate and remove the classic compiler logic?

Reply:
No, this was probably something I didn't refactor. We are already loading optional dependencies like `@remix-run/dev` dynamically so we can import these functions statically everywhere.